core/src/test/java/com/linecorp/armeria/client/retry/RetryingClientTest.java

```diff
@@ -672,7 +672,7 @@ public void onError(Throwable t) {}
                     public void onComplete() {}
                 });
 
-        await().untilAsserted(() -> assertThat(subscriberCancelServiceCallCounter.get()).isEqualTo(3));
+        await().untilAsserted(() -> assertThat(subscriberCancelServiceCallCounter.get()).isEqualTo(1));
```

Comment:
Could you explain why the value is changed from 3 to 1? 🤔

Reply:
I wanted to asked you about this 😅..
After this PR, `DeferredStreamMessages`'s `completionFuture` completes as soon as it is cancelled and this makes `RetryingClient` stop retrying as it's caught in [this line](https://github.com/line/armeria/blob/27f1d605bea0874bf0bd68962f5e405812ba55fd/core/src/main/java/com/linecorp/armeria/client/retry/RetryingClient.java#L259) in `RetryingClient`.

Reply:
According to https://github.com/line/armeria/pull/1242, it seems `RetryingClient` should stop retrying if either `HttpRequest` or `HttpResponse` gets aborted.

Is there a reason why it shouldn't do so when `subscription.cancel()` is called?
Currently, calling `DeferredStreamMesages#cancel` doesn't complete its `completionFuture`  and it passes the test.

Reply:
Yeah, that should be one, I guess. 😄 
Could you also change the name of the test, please?

Reply:
Fixed!
I've also chosen to not use `await().untilAsserted(...)` since this will make it always pass the test regardless of how many retry will actually happen.